runtime/j9vm/jvm.c

```diff
@@ -3989,15 +3989,79 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 			UDATA slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
 			Trc_SC_LoadLibrary_OpenShared(libName);
 
-/* jdk17+ calls JVM_LoadLibrary with decorated library names. If the following is done
- * then it overwrites the real error with a failure to load "liblib<name>.so.so".
- */
-#if JAVA_SPEC_VERSION < 17
 			if (0 != slOpenResult) {
-				slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags | J9PORT_SLOPEN_DECORATE);
-				Trc_SC_LoadLibrary_OpenShared_Decorate(libName);
+				char *libNameNotDecorated = (char *)libName;
+#if JAVA_SPEC_VERSION >= 17
+				/* JDK17+ jdk.internal.loader.NativeLibraries.load() calls JVM_LoadLibrary() with a decorated library name,
+				 * i.e., a path to the library name returned from Java_java_lang_System_mapLibraryName(nameNoPrefixNoExtension).
+				 * The library name passed to j9sl_open_shared_library() with the flag J9PORT_SLOPEN_DECORATE
+				 * has to be platform independent, i.e., it must not contain any prefix or file extension.
+				 */
+				char libPath[EsMaxPath] = {0};
+				const char *fileExt = strrchr(libName, '.');
+				BOOLEAN doOpenLibrary = TRUE;
+				if (NULL != fileExt) {
```

Comment:
Please invert `if` statements so both branches are textually near the condition:
```c
				if (NULL == fileExt) {
					/* A decorated library name is expected with a file extension,
					 * pass to j9sl_open_shared_library w/o modification.
					 */
					Trc_SC_libName_no_extension(libNameNotDecorated);
				} else {
```

Reply:
Updated